actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -231,6 +231,9 @@ class Parameters
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
     def initialize(parameters = {})
+      unless parameters.respond_to?(:with_indifferent_access)
```

Comment:
Hmm, I guess the complexity / cost of raising an `ArgumentError` is not worth the potential confusion of a `NoMethodError` raised since this is rarely used in a user code? 🤔